pettingzoo/utils/wrappers/terminate_illegal.py

```diff
@@ -39,7 +42,7 @@ def step(self, action):
             self.terminations[self.agent_selection]
             or self.truncations[self.agent_selection]
         ):
-            self._was_dead_step(action)
+            self._was_dead_step(None)
```

Comment:
I’m not 100% sure how the dead step functionality works but we should standardize all the code, as the examples and envs use was dead(action) but it might not be necessary

Reply:
The problem is action can be different than None even if truncations or terminations are True, so pyright gives an error. Possible solutions could be:

* The one I applied above
* Checking explicitly if action is None
* Ignoring the pyright error

Let me know what you prefer

Reply:
All of our example code uses these same three lines with dead step(action), is it possible to make the dead step function signature have type hints allowing None?

Reply:
I confirm that  `_was_dead_step(None)` is only used in the code above, so I reverted it to `_was_dead_step(action)` and ignored the pyright error.